### ISSUE TITLE
On the performance of KPConv

### ISSUE THREAD

**Issue by honggudrnjs** [HuguesTHOMAS/KPConv#134]
Hello, I am impressed by your work, but I have some questions on KPConv performance.
1. Performance Inconsistency
 In the KPConv paper, KPConv-deform (around 67% acc) performs better than KPConv-rigid (around 64% acc) on S3DIS dataset. However, according to [https://paperswithcode.com/sota/semantic-segmentation-on-s3dis] , KPConv with no prefix (I think it is KPConv-rigid, around 70%) performs better than KPConv-deform (around 67%). Because two different sources report two significantly different performances of KPConv-rigid, I am so confused. Could you describe why this happens? 

2. Reproducing the results
 I tried to reproduce by running PyTorch code on S3DIS dataset for 300 epochs, but interstingly, KPConv-rigid performs better than KPConv-deform, which is the opposite result compared to the result from the paper. Even, KPConv-rigid obtains around 67% accuracy and KPConv-deform obtains around 64% accuracy. Could you describe why this happens based on your experience?

3. Variability of performance 
 When I tried to reproduce, I observed that performances vary with a large gap (ranging from 55 to 65) during training. Therefore, I doubt that KPConv could achieve higher results by chance when performances vary during training. What do you think of wide performance range during training?

**Comment by HuguesTHOMAS** [comment]
Hi @honggudrnjs,

A lot of people focus on pure performances but forget that they do not mean everything. To answer your first question, refer to the paper and supplementary material. There are two different ways to evaluate scores on S3DIS, in a k-fold setup or only on Area-5. Depending on that, the scores are not the same.

For the PyTorch implementation, I focused on rigid KPConv and did not put as much effort into the deformable implementation because I am using this repository for other projects where I do not need the deform version. The PyTorch implementation does not reproduce exactly the Tensoflow implementation and I did not spend time to optimize parameters for better performances. It does not mean the deformable convolution is not good, it just was not pushed to its max potential. 

During training, the variability is inevitable as we evaluate a lot of different areas and they are not all equivalent. What matters is the voted results which are much more consistent and show that the method always reaches the same level of performance overall.
